sdk/tests/conformance2/transform_feedback/simultaneous_binding.html

```diff
@@ -0,0 +1,216 @@
+
+debug("");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+var gl = wtu.create3DContext(canvas, null, 2);
+
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+}
+
+function drawWithFeedbackBound(gl, prog, vao, tf, enableFeedback) {
+  gl.useProgram(prog);
+  gl.bindVertexArray(vao);
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+  if (enableFeedback) gl.beginTransformFeedback(gl.POINTS);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should not be any errors before drawing");
+  gl.drawArrays(gl.POINTS, 0, 4);
+  if (enableFeedback) gl.endTransformFeedback();
+  gl.bindVertexArray(null);
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+}
+
+function createBuffer(gl, dataOrSize) {
+  const buf = gl.createBuffer();
+  gl.bindBuffer(gl.ARRAY_BUFFER, buf);
+  gl.bufferData(gl.ARRAY_BUFFER, dataOrSize, gl.STATIC_DRAW);
+  gl.bindBuffer(gl.ARRAY_BUFFER, null);
+  return buf;
+}
+
+function createVAO(gl, buf, inLoc) {
+  const vao = gl.createVertexArray();
+  gl.bindVertexArray(vao);
+  gl.bindBuffer(gl.ARRAY_BUFFER, buf);
+  gl.enableVertexAttribArray(0);
+  gl.vertexAttribPointer(0, 1, gl.FLOAT, false, 0, 0);
+  gl.bindBuffer(gl.ARRAY_BUFFER, null);
+  gl.bindVertexArray(null);
+  return vao;
+}
+
+const prog = wtu.setupTransformFeedbackProgram(gl, ["vshader", "fshader"],
+    ["out_value"], gl.SEPARATE_ATTRIBS,
+    ["in_value"]);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "linking transform feedback shader should not set an error");
+
+const vertexBuffer = createBuffer(gl, new Float32Array([1, 2, 3, 4]));
+const vao = createVAO(gl, vertexBuffer);
+
+const tfBuffer = createBuffer(gl, new Float32Array([0, 0, 0, 0]));
+
+const tf = gl.createTransformFeedback();
+gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+gl.useProgram(prog);
+gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer);
+// this binds the default (id = 0) TRANSFORM_FEEBACK buffer
+gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+const uniformBuffer = createBuffer(gl, new Float32Array([1, 0, 0, 0]));
+const ubi = gl.getUniformBlockIndex(prog, "UniformBlock");
+gl.uniformBlockBinding(prog, ubi, 0);
+gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, uniformBuffer);
+
+drawWithFeedbackBound(gl, prog, vao, tf, true);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "transform feedback should be successful");
+
+const expected = [2, 4, 6, 8];
+gl.bindBuffer(gl.TRANSFORM_FEEDBACK_BUFFER, tfBuffer);
+wtu.checkFloatBuffer(gl, gl.TRANSFORM_FEEDBACK_BUFFER, expected);
+
+debug("<hr/>Test ARRAY_BUFFER");
+// this should fail because the transform feedback's buffer #0 and the
+// badVao's buffer #0 are the same buffer
+const badVao = createVAO(gl, tfBuffer);
+drawWithFeedbackBound(gl, prog, badVao, tf, false);
+wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as vertex attrib and tf simultaneously");
```

Comment:
Most call sites of drawWithFeedbackBound in this test make two calls, with "true" and "false" as the last argument. Suggest adding another function which makes the two calls internally, so you can avoid duplicating the detail strings. This would also make it easier to add a suffix like "(with tf enabled)" or similar.